tests/unit/s2n_async_pkey_test.c

```diff
@@ -142,6 +152,120 @@ int async_pkey_store_callback(struct s2n_connection *conn, struct s2n_async_pkey
     return S2N_SUCCESS;
 }
 
+int async_pkey_signature_callback(struct s2n_connection *conn, struct s2n_async_pkey_op *op)
+{
+    pkey_op = op;
+
+    s2n_async_pkey_op_type type = { 0 };
+    EXPECT_SUCCESS(s2n_async_pkey_op_get_op_type(op, &type));
+    EXPECT_EQUAL(type, S2N_ASYNC_SIGN);
+
+    uint8_t expected_size = 0;
+    EXPECT_SUCCESS(s2n_hash_digest_size(S2N_HASH_SHA1, &expected_size));
+
+    uint32_t input_size = 0;
+    EXPECT_SUCCESS(s2n_async_pkey_op_get_input_size(op, &input_size));
+    EXPECT_EQUAL(input_size, expected_size);
+
+    struct s2n_blob input = { 0 };
+    EXPECT_SUCCESS(s2n_alloc(&input, input_size));
+
+    struct s2n_blob expected_digest = { 0 };
+    EXPECT_SUCCESS(s2n_alloc(&expected_digest, expected_size));
+
+    struct s2n_hash_state digest = { 0 };
+    EXPECT_SUCCESS(s2n_hash_new(&digest));
+    EXPECT_SUCCESS(s2n_hash_init(&digest, S2N_HASH_SHA1));
+    EXPECT_SUCCESS(s2n_hash_update(&digest, test_digest_data, test_digest_size));
+    EXPECT_SUCCESS(s2n_hash_digest(&digest, expected_digest.data, expected_digest.size));
+    EXPECT_SUCCESS(s2n_hash_free(&digest));
+
+    EXPECT_SUCCESS(s2n_async_pkey_op_get_input(op, input.data, input.size));
+    EXPECT_BYTEARRAY_EQUAL(input.data, expected_digest.data, expected_digest.size);
+    EXPECT_SUCCESS(s2n_free(&input));
```

Comment:
Should you need to free the input? Or should that be handled by the op?

Reply:
The op doesn't allocate memory to contain input, it just copies whatever it has into the user's buffer.